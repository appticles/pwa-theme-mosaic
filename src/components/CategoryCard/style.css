.item {
  position: relative;
}

/* override semantic ui styles */
/* item header */
.ui.header {
  position: absolute;
  z-index: 1;
  top: 10%;
  left: 0;
  right: 0;
}
/* image */
.ui.image.big {
  width: 100vw;
}
