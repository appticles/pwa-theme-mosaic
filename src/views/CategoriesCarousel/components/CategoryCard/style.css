.categories-card .item {
  position: relative; }

.categories-card .ui.header {
  z-index: 1;
  display: flex;
  align-items: flex-end;
  flex-direction: row;
  text-align: start;
  padding: 9px 15px;
  background: -webkit-linear-gradient(bottom, rgba(0, 0, 0, 0.4), rgba(0, 0, 0, 0.08) 55px, transparent 80px, transparent); }
  .categories-card .ui.header h1 {
    font-weight: 100;
    text-transform: uppercase;
    color: #ffffff;
    overflow-wrap: break-word;
    max-width: 100%; }

.categories-card .ui.header.header > div {
  height: unset;
  overflow: hidden;
  text-overflow: ellipsis; }

/*
.categories .ui.image.big {
  width: 100vw;
}
*/
.categories-card .item {
  background-repeat: no-repeat;
  background-size: cover;
  background-color: #32394a; }
