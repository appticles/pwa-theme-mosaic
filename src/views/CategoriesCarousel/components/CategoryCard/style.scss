@import "../../../../utilities/variables";

.categories-card .item {
  position: relative;
}

.categories-card .ui.header {
  position: absolute;
  z-index: 1;
  top: 0;
  left: 0;
  right: 0;
  display: flex;
  align-items: flex-end;
  text-align: start;
  font-weight: 100;
  text-transform: uppercase;
  padding: 9px 15px;
  background: -webkit-linear-gradient(
    bottom,
    rgba(0, 0, 0, 0.4),
    rgba(0, 0, 0, 0.08) 55px,
    transparent 80px,
    transparent
  );
  h1 {
    color: $cover-text-color;
  }
}

.categories-card .ui.header.header > div {
  height: unset;
  overflow: hidden;
  text-overflow: ellipsis;
}

/*
.categories .ui.image.big {
  width: 100vw;
}
*/

.categories-card .item {
  background-repeat: no-repeat;
  background-size: cover;
  background-color: $cover-panel-color;
}
