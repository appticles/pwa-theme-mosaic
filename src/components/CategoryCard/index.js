import React, { Component } from 'react';
import { Item, Header } from 'semantic-ui-react';
import { Link } from 'react-router-dom';

import { categoryPropType } from '../../views/CategoriesCarousel/reducer';
import './style.css';

export default class CategoryCard extends Component {
  render() {
    const { category } = this.props;
    return (
      <Link to={'/category/' + category.slug + '/' + category.id}>
        <Item>
          <Item.Header as={Header}>{category.name}</Item.Header>
        </Item>
      </Link>
    );
  }
}

CategoryCard.propTypes = {
  category: categoryPropType.isRequired,
};
CategoryCard.defaultProps = {
  tall: 0,
};
