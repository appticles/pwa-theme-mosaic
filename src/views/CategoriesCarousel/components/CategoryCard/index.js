import React from 'react';
import { Item, Header } from 'semantic-ui-react';
import { Link } from 'react-router-dom';

import { categoryPropType } from '../../reducer';
import './style.css';

<<<<<<< HEAD
const CategoryCard = (props) => {
  const { slug, id, name } = props.category;
  return (
    <Link to={'/category/' + slug + '/' + id}>
      <Item>
        <Item.Header as={Header}>
          <div dangerouslySetInnerHTML={{ __html: name }} />
        </Item.Header>
      </Item>
    </Link>
  );
};
=======
const CategoryCard = props => (
  <Link to={'/category/' + props.category.slug + '/' + props.category.id}>
    <Item>
      <Item.Header as={Header}>
        <h2 dangerouslySetInnerHTML={{ __html: props.category.name }} />
      </Item.Header>
    </Item>
  </Link>
);
>>>>>>> fb3683ab5c649ec7e4f3a29779717db044ce5ee0

CategoryCard.propTypes = {
  category: categoryPropType.isRequired,
};

export default CategoryCard;
