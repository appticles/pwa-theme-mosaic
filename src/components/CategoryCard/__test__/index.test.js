import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import CategoryCard from '../index';

describe('Category card', () => {
  it('should render correctly', () => {
    const mockCategory = {
      slug: 'mock-slug',
      id: 1,
      name: 'Some mock name',
    };
    const output = shallow(<CategoryCard category={mockCategory} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
