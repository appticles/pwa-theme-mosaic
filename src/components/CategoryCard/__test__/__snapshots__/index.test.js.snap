// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category card should render correctly 1`] = `
<Link
  replace={false}
  to="/category/mock-slug/1"
>
  <Item>
    <ItemHeader
      as={[Function]}
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "Some mock name",
          }
        }
      />
    </ItemHeader>
  </Item>
</Link>
`;
